src/gil.rs

```diff
@@ -358,13 +361,21 @@ impl Drop for LockGIL {
 
 ///
 /// [Memory Management]: https://pyo3.rs/main/memory.html#gil-bound-memory
+#[cfg_attr(
+    not(feature = "gil-refs"),
+    deprecated(
+        since = "0.21.0",
+        note = "`GILPool` will have no function after PyO3's GIL Refs API is removed"
```

Comment:
I would make this into something that is actionable in 0.21.0, i.e. `GILPool` does have no function if the GIL Refs API is not used.